### DISCUSSION TITLE
Get arguments passed in script?

### DISCUSSION THREAD

**Discussion by trymeouteh** [nushell/nushell#13575] [Q&A] [answered]
How do you read the passed arguments in a Nushell script?

```
./script.nu alpha
./script.nu bravo
```

```
#!/usr/bin/env nu

if $1 == 'alpha' {
    print 'alpha'
} else if $2 == 'bravo' {
    print 'bravo'
}
```

**Comment by Lockszmith-GH** [comment, marked as answer]
https://www.nushell.sh/book/scripts.html#parameterizing-scripts

**Reply by trymeouteh** [reply]
Can you use parameters without having a main function?

**Reply by fdncred** [reply]
I don't think so. `main` is the entry point for these type of scripts. $1 $2 is not a thing in nushell.

**Reply by trymeouteh** [reply]
I do not understand `def main()` compared to `fn main()`. `fn main()` does not work but `def main()` does work.

**Reply by fdncred** [reply]
There's no such thing as `fn` in nushell. Custom commands begin with `def`.

**Reply by Lockszmith-GH** [reply]
[def](https://www.nushell.sh/commands/docs/def.html#def-for-core)

**Reply by Lockszmith-GH** [reply]
@trymeouteh where are you basing your source on? are you using some examples? did you draw them from the documentation?
If so, please send links to source.

I highly recommend reviewing the documentation, while it's not complete, it's still an excellent source that should be made better - if you did encounter mistakes, pointing at those mistakes would go a long way towards fixing them

**Reply by trymeouteh** [reply]
I got the `fn` from Brave Search AI which turned out to be wrong.

So then `def` is the same or simular to `function` or `func` to create a function in Nushell?

**Reply by fdncred** [reply]
`def` means definition of a nushell custom command. it's mostly synonymous with function in another language.

**Reply by trymeouteh** [reply]
Thank you everyone for explaining to me how to get arguments passed into the scripts and how `def` works.

**Reply by Lockszmith-GH** [reply]
Ah, AI ! those lying entities in the cloud. Never trust an AI I say, you don't know where they aligence stand.
joking aside.
AI is a great first place to learn something, but when something doesn't work - I highly recommend you check out the docs yourself.

Especially when approaching a community, make the effort - it's worth it. You'll really appreciate it as you become part of the of the community too.

In anyway, welcome to the wonderful world of 'figuring nu out'. I'm new here myself, and I can attest, awesome human beings here helping out each others.
